Modules/SubModule_Import/xASL_imp_NII2BIDS_Run.m

```diff
@@ -1,59 +1,89 @@
 %
 % -----------------------------------------------------------------------------------------------------------------------------------------------------
-% EXAMPLE:     x = xASL_imp_NII2BIDS_Run(x, bidsPar, studyPar, subjectSessionLabel, inSessionPath, outSessionPath, listRuns, iRun);
+% EXAMPLE:     x = xASL_imp_NII2BIDS_Run(x, imPar, bidsPar, studyPar, listRuns, nameSubjectSession, bidsLabel, iRun);
 % __________________________________
 % Copyright 2015-2021 ExploreASL
 
+    %% Print the run that is being converted
+    fprintf('\n[\b======================================== CONVERT RUN =========================================]\b\n');
+    fprintf('Converting subject %s, session %s, run %s ', bidsLabel.subject, bidsLabel.visit, listRuns{iRun});
 
-    %% Anatomical files
-    fprintf('run %d ...\n', iRun);
-    try
-        xASL_imp_NII2BIDS_RunAnat(imPar, bidsPar, studyPar, subjectSessionLabel, outSessionPath, listRuns, iRun, nameSubjectSession);
-    catch loggingEntry
-        [x] = xASL_qc_AddLoggingInfo(x, loggingEntry);
-        xASL_imp_NII2BIDS_RunIssueWarning(loggingEntry, 'anatomical', subjectSessionLabel, iRun);
-    end
-    
-    %% Perfusion files
-    try
-        xASL_imp_NII2BIDS_RunPerf(imPar, bidsPar, studyPar, subjectSessionLabel, inSessionPath, outSessionPath, listRuns, iRun);
-    catch loggingEntry
-        [x] = xASL_qc_AddLoggingInfo(x, loggingEntry);
-        xASL_imp_NII2BIDS_RunIssueWarning(loggingEntry, 'perfusion', subjectSessionLabel, iRun);
+    %% 1. Make a subject directory with a correct session name
+    if ~isempty(bidsLabel.visit)
+        %if ~isempty(bidsLabel.visit)
+        sessionLabel = ['ses-' bidsLabel.visit];
+        %else
+        %    sessionLabel = ['ses-' listRuns{iRun}(5:end)];
+        %end
+        
+        % Create directory if it does not exist already
+        sessionPerfusionDirectory = fullfile(imPar.BidsRoot,['sub-' bidsLabel.subject], sessionLabel,'perf');
+        xASL_adm_CreateDir(sessionPerfusionDirectory);
+        
+        inSessionPath = fullfile(imPar.TempRoot, nameSubjectSession, listRuns{iRun});
+        outSessionPath = fullfile(imPar.BidsRoot, ['sub-' bidsLabel.subject], sessionLabel);
+
+        % Need to add the underscore so that it doesn't need to be added automatically and can be skipped for empty session
+        sessionLabel = ['_' sessionLabel];
+    else
+        % Session label is skipped
+        sessionLabel = '';
+
+        % Only one session - no session labeling
+        if ~exist(fullfile(imPar.BidsRoot,['sub-' bidsLabel.subject]),'dir')
+            xASL_adm_CreateDir(fullfile(imPar.BidsRoot, ['sub-' bidsLabel.subject], bidsPar.strPerfusion));
+        end
+        inSessionPath = fullfile(imPar.TempRoot, nameSubjectSession, listRuns{iRun});
+        outSessionPath = fullfile(imPar.BidsRoot, ['sub-' bidsLabel.subject]);
     end
-    
-end
 
 
+    %% 2. Convert structural and ASL runs
+	subjectSessionLabel = [bidsLabel.subject sessionLabel];
+	try
```

Comment:
These try-catches should now be in `xASL_Iteration` right?

Reply:
Hm... do they hurt here? Logging info and everything is called and this is also beneficial if there is only a single corrupted scan.

Reply:
Well the more try-catches the more debugging becomes a pain, as it won't print the original error or warning right?
The single corrupted scan you want to see in the general iteration on the subject level. Unless this is a very specific try-catch structure that you created for your separate logging?

Reply:
Actually, catching closer to the error might mean more information. And Michael made a dedicated logging. So that might actually be what @HenkMutsaerts wants. @MichaelStritt am I right?

Reply:
More information in the specific log, but less information in the screen output and diary log. So if you just want to scroll through the diary and see what goes wrong, this is a pain. And I still think it is nicer to scroll through the diary, which gives you the context of where the error or warning happens, then having the error only in the dedicated logging, which just mentions the line number. So this is nicer when developing and running a single subject.

But Michael's method is of course much nicer if you iterate through 100 subjects.

Is there a way to do both? Perhaps to always print the output of the catched error on the screen as well?

Also, it is important that this happens the same way for both import and processing modules. We should avoid this:
`if module==import, then ... if module==processing, then ...`

Reply:
> 
> 
> Actually, catching closer to the error might mean more information. And Michael made a dedicated logging. So that might actually be what @HenkMutsaerts wants. @MichaelStritt am I right?

I like my new logging features, but that's personal preference I guess.